### ISSUE TITLE
Let forbidden be forbidden

### ISSUE THREAD

**Issue by karex** [Arsenal21/all-in-one-wordpress-security#92]
There are some places, where the AIOWPS dies to prevent malicious requests and user actions, and these are handled with `wp_die`, but without the `$args` argument filled, and that yields a HTTP status code of 500, which pollutes the server's logs, and one might think that something is wrong, which is not the case.

These situations should return a 403 status code in my understanding.

An easy fix would be to specify status 403 when wp_die is called (maybe with a filterable constant). Additionally, I think it would be great to be able to filter these "dies", because e.g. one might want to redirect these requests instead of displaying errors. For this, a "framing function" for die events and a filter might be great, where we might be able to filter the actual function called in these "die" events.

What do you think?

**Comment by Arsenal21** [comment]
I agree. However, we can't make a global change like this. It needs to be looked at on a case by case basis.  So you have to be more specific in terms of which exact instance (or instances) you are talking about. Then we can look at those instances to see if there is any other ramifications and then make changes accordingly.

**Comment by karex** [comment]
Yeah, true, there are different places where different HTTP status codes might be appropriate.

What concerned me actually is where [Stop User Enumeration](https://github.com/Arsenal21/all-in-one-wordpress-security/blob/3a8874196ae704118cf1ab066190ce420dd7e2c6/all-in-one-wp-security/other-includes/wp-security-stop-users-enumeration.php) is implemented. Here a 403 would be preferable probably.

Other places might need a review too. For example think the [locked out IPs](https://github.com/Arsenal21/all-in-one-wordpress-security/blob/c9c20fa453ace2e4c793f3baab9070d7edcc3e4d/all-in-one-wp-security/classes/wp-security-user-login.php) might have a status code of 403 or 404.

I conclude this from [RFC 2616 10.4.4](https://tools.ietf.org/html/rfc2616#section-10.4.4), which says that 403 is for events when "server understood the request, but is refusing to fulfill it.", but if the server does not want to make the reason of the refusal available, then "the status code 404 (Not Found) can be used instead."
